### PR TITLE
Add flagutil.Duration to avoid conversion bugs

### PR DESCRIPTION
**Pull request by dima-vm**
I had a bug incorrectly using `.Msec`, when converting it to time.Duration.
Added flagutil.Duration() which eliminates the possibility of incorrectly using the `.Msec`.

### CODE REVIEW

**Review by hagen1778 — COMMENTED** [review]
It is unclear what exactly all these changes are fixing in the code. I'm not against adding a .Duration method, though.

The commits are unsigned. The linter is failing.

**Comment by f41gh7** [comment]
I think, we have to re-check all timestamps usage at our code base and convert integer timestamps into typed format with time.Time or time.Duration types. Time manipulation must be executed only with time.Add or similar methods.

 It must prevent time conversion bugs. Such as recent storage bug, that cause panic and data loss at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4877

 I suggest to create an issue for code refactoring. 

Following PR fixes only small part of possible problems, it must be updated with necessary changes.

**Comment by dima-vm** [comment]
@hagen1778 I didn't post a bug, because resolved it in the same PR. Basically, forgot to divide/multiply `Msec` by `time.Millisecond` as I felt it's already in milliseconds. Basically, bad readability makes is more bug-prone.

Typed Duration would avoid that issue. I'd even go further and make `Msec` private, refactoring all its usages. Totally agree with @f41gh7 basically.

**Comment by dima-vm** [comment]
@hagen1778 @f41gh7 PTAL.

There's only one usage of `.Msecs` left, in [vmstorage/main.go#L78](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/a20234d2d056d0467727fc3f501565e2a4f53b7c/app/vmstorage/main.go#L78). But I think it's better resolved in a follow-up PR, by using time.Time, as @f41gh7 noted above.

**Comment by hagen1778** [comment]
@dlazerka commits are still unsigned

File: lib/promutils/duration.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func NewDuration(d time.Duration) *Duration {
 }
 
 // MarshalYAML implements yaml.Marshaler interface.
-func (pd Duration) MarshalYAML() (interface{}, error) {
+func (pd *Duration) MarshalYAML() (interface{}, error) {
```
Comment (hagen1778) [inline lib/promutils/duration.go:22]: why changing this?

File: lib/storage/storage.go [hunk 2]
```diff
@@ -246,7 +243,7 @@ func MustOpenStorage(path string, retentionMsecs int64, maxHourlySeries, maxDail
 
 	// Initialize nextRotationTimestamp
 	nowSecs := time.Now().UnixNano() / 1e9
-	nextRotationTimestamp := nextRetentionDeadlineSeconds(nowSecs, retentionMsecs/1000, retentionTimezoneOffsetSecs)
+	nextRotationTimestamp := nextRetentionDeadlineSeconds(nowSecs, retention.Milliseconds()/1000, retentionTimezoneOffsetSecs)
```
Comment (hagen1778) [inline lib/storage/storage.go:246]:
```suggestion
	nextRotationTimestamp := nextRetentionDeadlineSeconds(nowSecs, retention.Second(), retentionTimezoneOffsetSecs)
```

File: lib/storage/storage.go [hunk 3]
```diff
@@ -33,8 +33,8 @@ import (
 )
 
 const (
-	msecsPerMonth     = 31 * 24 * 3600 * 1000
-	maxRetentionMsecs = 100 * 12 * msecsPerMonth
+	retention31Days = 31 * 24 * time.Hour
```
Comment (hagen1778) [inline lib/storage/storage.go:36]: `retention31Days` has different meaning to `retention1Month`. 
In our docs, we operate with `month` unit. Would be nice to have constant name consistent with that

**Comment by hagen1778** [comment]
> It must prevent time conversion bugs. Such as recent storage bug, that cause panic and data loss at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4877

I don't think that using time conversion would protect from that. Thorough unittest could.

File: lib/storage/storage.go (hunk 2 above)
Comment (dima-vm) [reply lib/storage/storage.go]: No, the Seconds() returns float64, and does more things inside (although pretty fast probably, but still unnecessary conversion to float and back).

**Comment by dima-vm** [comment]
Adressed all review comments, PTAL

File: lib/storage/storage.go (hunk 2 above)
Comment (dima-vm) [reply lib/storage/storage.go]: Added comment to code

**Review by dima-vm — COMMENTED** [review]
PTAL, Split "31days" stuff to a separate PR as requested by @valyala

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@dima-vm , thanks for the pull request!

**Pull request merged**